src/storage/src/source/postgres/replication.rs

```diff
@@ -211,10 +211,19 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
             while let Some(event) = rewind_input.next().await {
                 if let AsyncEvent::Data(caps, data) = event {
                     for req in data {
-                        assert!(
-                            resume_lsn <= req.snapshot_lsn + 1,
-                            "slot compacted past snapshot point. snapshot consistent point={} resume_lsn={resume_lsn}", req.snapshot_lsn + 1
-                        );
+                        if resume_lsn > req.snapshot_lsn + 1 {
+                            let err = DefiniteError::SlotCompactedPastResumePoint(req.snapshot_lsn + 1, resume_lsn);
```

Comment:
Thank you so much for moving this error in here. I'd love to see if there's a way we can unify the return of these definite errors, but that's a totally non-blocking request.